presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java

```diff
@@ -70,7 +72,7 @@ public void testAcos()
     {
         for (double doubleValue : DOUBLE_VALUES) {
             assertFunction("acos(" + doubleValue + ")", DOUBLE, Math.acos(doubleValue));
-            assertFunction("acos(FLOAT '" + (float) doubleValue + "')", DOUBLE, Math.acos((float) doubleValue));
+            assertFunction("acos(FLOAT '" + doubleValue + "')", DOUBLE, Math.acos((float) doubleValue));
```

Comment:
This change introduces inconsistency. You build FLOAT value in SQL based on `Double.toString()` while comparing result to `acos(float)` when float is casted from double value by Java casting. I'd keep `(float) doubleValue` in FLOAT c-tor for consistency. This test should not check constructing FLOAT out of Double.